cpp-client-reference/CMakeLists.txt

```diff
@@ -6,17 +6,41 @@ set(CMAKE_TOOLCHAIN_FILE
 
 project(volition)
 
+include(CheckPIESupported)
+check_pie_supported()
+
 find_package(gRPC CONFIG REQUIRED)
 find_package(Protobuf REQUIRED)
 
 set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+add_library(common_config INTERFACE)
+target_compile_definitions(common_config INTERFACE
+    $<$<CXX_COMPILER_ID:Clang,GNU>:
```

Comment:
oh help, what is `$<$<CXX:Clang,GNU>:` do?

Reply:
It means the following flags are only used when compiling with Clang or GCC. These flags are specific to these compilers.

https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html